### PR TITLE
Issue #6402: add new SuppressionSingleFilter

### PR DESCRIPTION
**Pull request by rnveach**
Issue #6402

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionSingleFilter.java [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+    /**
+     * Specify a comma-separated list of values, where each value is an integer or a range of
+     * integers denoted by integer-integer.
+     */
+    private String columns;
+
+    /**
+     * Setter to define the RegExp for matching against the file name associated with an audit
+     * event.
+     * @param files the name of the file
```
[the remark this answers is not in the document]
Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionSingleFilter.java]: The parameter you mention is in a different class. Those parameters are part of the new, second constructor and the description for those parameters match the 1st constructor.
So you want the new parameters to match the new filter but not match the old constructor?

[the remark this answers is not in the document]
Comment (pbludov) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionSingleFilter.java]: I just noted that the parameter name is `files` but the description is `the name of the file`.

[the remark this answers is not in the document]
Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionSingleFilter.java]: @pbludov Is that better?

**Comment by romani** [comment]
reminder to test code by https://github.com/sevntu-checkstyle/methods-distance

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionSingleFilter.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (pbludov) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionSingleFilter.java]:
```
@param files regular expression for filtered file names
```
is better then
```
@param files the name of the file
```

**Comment by rnveach** [comment]
> reminder to test code by https://github.com/sevntu-checkstyle/methods-distance

`SuppressionSingleFilter` has a "Penalty function value: 0.0".

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionSingleFilter.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionSingleFilter.java]: Done.

**Comment by romani** [comment]
@pbludov , please finish review

**Pull request merged**